### ISSUE TITLE
Feature Request: Option to choose between regular LLM chat, and Notes- Chat + Custom "prompt template"

### ISSUE THREAD

**Issue by FemBoxbrawl** [reorproject/reor#101]
First for the different chat options, being able to choose between a regular LLM chat and a Notes chat would be useful, as when you want the LLM to generate ideas, like brainstorming, without having conflicting notes changing the context of the answer. And if you want a chat focused on notes, then NOTE-chat should be enabled (what we have right now)

And what also would be cool is a "prompt template" box, or "instruction box" to give custom instructions to an LLM to make it act a certain way, etc. Great App, by the way!

**Comment by samlhuillier** [comment]
You mean you'd want to choose between a normal llm (like in chatgpt) and an llm that has been given your notes as context? We are working on features related to customising the prompt configs as we speak.

**Comment by FemBoxbrawl** [comment]
> You mean you'd want to choose between a normal llm (like in chatgpt) and an llm that has been given your notes as context? We are working on features related to customising the prompt configs as we speak.

essentially what i mean is a RAG on / RAG off button, so yeah. But great to hear!  can't wait

**Comment by Arcovoltaire** [comment]
I believe he touched on an interesting point: The possibility of chatting only with the active note. Or for example with notes selected, it could be like: Nodes within a folder.

**Comment by chuyouchia** [comment]
> I believe he touched on an interesting point: Adding the possibility of chatting only with the active note. Or for example with notes selected, it could be like: Notes within a folder.

Yes, this is coming in the next release. Do you have specific use cases or situations you anticipate using note-specific chat? Want to understand more so that as we develop this, we can keep the use case in mind!

**Comment by samlhuillier** [comment]
This is now out! @Arcovoltaire @FemBoxbrawl In the [latest release](https://github.com/reorproject/reor/releases/tag/v0.1.53) you can chat with the current note like this:

https://github.com/reorproject/reor/assets/17236551/3806911b-e95c-42fb-9b2c-18268a817f9d

**Comment by FemBoxbrawl** [comment]
Lets go! thanks so much!

**Comment by Arcovoltaire** [comment]
> This is now out! @Arcovoltaire @FemBoxbrawl In the [latest release](https://github.com/reorproject/reor/releases/tag/v0.1.53) you can chat with the current note like this:

Very good! It's a tremendous addition to the program. Congratulations!!!

I did a more detailed analysis here:
[https://github.com/reorproject/reor/issues/118](url)
